### PR TITLE
Remove interfaceID from PnP API client API & helpers

### PR DESCRIPTION
**Pull request by jspaith**
Per #1528, we have interfaceId in samples/test/API shape but this is NOT needed by the protocol between the device and hub anymore and was effectively a giant no-op.  Shout out to @rido-min for capturing this.

This change:
* Removes `interfaceId` from the API shape and most of the logic in core code dealing with it.
* Addresses an issue where we used the words "interfaceName" quite a bit.  What we really meant was component or componentName.  Fixed up.

NOTE::
There will be some followup work after this change.  In particular see #1561; we're not consistent pre-checking modelId and componentName.  If we decide to pre-check nothing (I think in new protocol design this is much more palatable, following up) then #1561 would also mean we'd remove `DT_InterfaceClient_CheckInterfaceIdValid`.  I don't want to remove this one yet, because if we do want to pre-parse modelID we'd just use `DT_InterfaceClient_CheckInterfaceIdValid` and I'm not yanking till we're sure either way.  The GitHub issue will act as our IOU to do this if need-be.

### CODE REVIEW

File: digitaltwin_client/samples/digitaltwin_sample_environmental_sensor/digitaltwin_sample_environmental_sensor.c [hunk 1]
```diff
@@ -1,11 +1,10 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-// Implements a sample DigitalTwin interface that integrates with an environmental sensor (for reporting 
-// temperature and humidity over time).  It also has basic commands and properties it can process,
-// such as setting brightness of a light and blinking LEDs.  Because this sample is designed
-// to be run anywhere, all of the sameple data and command processing is expressed simply with 
-// random numbers and LogInfo() calls.
+// Implements a sample DigitalTwin interface as defined by "dtmi:my_company:com:EnvironmentalSensor;1"
```
Comment (danewalton) [inline digitaltwin_client/samples/digitaltwin_sample_environmental_sensor/digitaltwin_sample_environmental_sensor.c:4]: does the user know where to find this `dtmi:my_company:com:EnvironmentalSensor;1`?
at least the json document in this directory has a value named `dtmi:com:example:EnvironmentalSensor;1`

File: digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c [hunk 2]
```diff
@@ -1,9 +1,9 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-// Implements a sample DeviceInfo DigitalTwin interface.  As the name implies, this interface returns
-// information about the device such as the OS version, amount of storage, etc.
-//
+// Implements a sample DeviceInfo DigitalTwin interface of dtmi:azure:DeviceManagement:DeviceInformation;1.
```
Comment (danewalton) [inline digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c:4]: Is the JSON representing `dtmi:azure:DeviceManagement:DeviceInformation;1` missing from this directory?

File: digitaltwin_client/inc/internal/dt_interface_private.h [hunk 3]
```diff
@@ -33,11 +33,9 @@ typedef enum DT_COMMAND_PROCESSOR_RESULT_TAG
 #define DT_COMMAND_ERROR_STATUS_CODE  500
 
 MOCKABLE_FUNCTION(, int, DT_InterfaceClient_CheckComponentNameValid, const char*, componentName);
-MOCKABLE_FUNCTION(, int, DT_InterfaceClient_CheckInterfaceIdValid, const char*, interfaceName);
+MOCKABLE_FUNCTION(, int, DT_InterfaceClient_CheckInterfaceIdValid, const char*, interfaceId);
```
[the remark this answers is not in the document]
Comment (jspaith) [reply digitaltwin_client/inc/internal/dt_interface_private.h:36]: `DT_InterfaceClient_CheckInterfaceIdValid` at some point *might* be used to check the `modelId` parameter passed in by `DigitalTwin_DeviceClient_CreateFromDeviceHandle`.  This is GitHub issue I reference in comments.

If we don't do this checking after all, I'll remove this function.  If we need it, then I'll wire function up.  I didn't want to remove this function yet only to have to add it back, though.

File: digitaltwin_client/samples/digitaltwin_sample_environmental_sensor/digitaltwin_sample_environmental_sensor.c (hunk 1 above)
Comment (jspaith) [reply digitaltwin_client/samples/digitaltwin_sample_environmental_sensor/digitaltwin_sample_environmental_sensor.c:4]: Yeah - that mismatch is no good.  I'm updating .c to reflect the JSON in dir.  Ultimately this JSON will be in a central spot all SDKs can point to (#1529 tracks) so the long-term documenting of this only will be solved once we get a spot to point at but not until then.

File: digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c (hunk 2 above)
Comment (jspaith) [reply digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c:4]: Yes.  Per #1529 the DeviceInfo needs a permanent spot we can link to.  Note that we checked the sample JSON into our C SDK because its samples, whereas DeviceInfo is a "system" interface we didn't want to checkin.  I get its inconsistent which is why we have the IOU bug.

File: digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
+// As the name implies, this interface returns information about the device such as the OS version, amount of storage, etc.
```
Comment (ericwol-msft) [inline digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c:5]: too wordy

```suggestion
// This interface returns information about the device such as the OS version, amount of storage, etc.
```

File: digitaltwin_client/samples/digitaltwin_sample_environmental_sensor/digitaltwin_sample_environmental_sensor.c [hunk 5]
```diff
@@ -1,11 +1,10 @@
...
+// This demonstrates the core DigitalTwin concepts of telemetry, commands, and properties.
```
Comment (ericwol-msft) [inline digitaltwin_client/samples/digitaltwin_sample_environmental_sensor/digitaltwin_sample_environmental_sensor.c:5]:
```suggestion
// This demonstrates the core Azure Digital Twin concepts of telemetry, commands, and properties.
```

Comment (jspaith) [reply digitaltwin_client/samples/digitaltwin_sample_environmental_sensor/digitaltwin_sample_environmental_sensor.c:5]: Fixing.

File: digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c (hunk 4 above)
Comment (jspaith) [reply digitaltwin_client/samples/digitaltwin_sample_device_info/digitaltwin_sample_device_info.c:5]: As the code I shall soon checkin implies, I have fixed this.
